contrib/pinot-druid-migration/src/main/java/org/apache/pinot/druid/tools/DruidToPinotSchemaGenerator.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.druid.tools;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.io.FileUtils;
+import org.apache.druid.jackson.DefaultObjectMapper;
+import org.apache.druid.query.DruidProcessingConfig;
+import org.apache.druid.segment.IndexIO;
+import org.apache.druid.segment.QueryableIndex;
+import org.apache.druid.segment.QueryableIndexStorageAdapter;
+import org.apache.druid.segment.column.ColumnCapabilities;
+import org.apache.druid.segment.column.ColumnConfig;
+import org.apache.druid.segment.column.ColumnHolder;
+import org.apache.druid.segment.column.ValueType;
+import org.apache.druid.segment.data.Indexed;
+import org.apache.pinot.common.data.FieldSpec;
+import org.apache.pinot.common.data.Schema;
+
+/**
+ * The DruidToPinotSchemaConverter is a tool that takes a Druid segment and creates a Pinot schema with the
+ * segment's information.
+ */
+public class DruidToPinotSchemaGenerator {
```

Comment:
The input for schema converter should not be `druid segment`. It should be `ingestion spec` https://druid.apache.org/docs/latest/ingestion/index.html#ingestion-specs

Reply:
It turns out that the Druid ingestion spec does not have enough info to generate a Pinot schema from it. It does not specify which columns are single/multi-valued, so I had to go ahead and use this original method anyway. I only create an `adapter` from the segment, and I don't have to read any rows.